webapp/src/test/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartlingTest.java

```diff
@@ -1684,6 +1741,338 @@ public void testBatchesFor() {
+      assetExtractionService.createAssetTextUnit(assetExtraction, name, content, comment);
+
+      tmService.addTMTextUnitCurrentVariant(
+          tu.getId(),
+          frCA.getId(),
+          String.format("%s in %s", content, frCA.getBcp47Tag()),
+          "test comment",
+          TMTextUnitVariant.Status.MT_TRANSLATED);
+      tmService.addTMTextUnitCurrentVariant(
+          tu.getId(),
+          jaJP.getId(),
+          String.format("%s in %s", content, jaJP.getBcp47Tag()),
+          "test comment",
+          TMTextUnitVariant.Status.MT_TRANSLATED);
+
+      ThirdPartyTextUnit thirdPartyTextUnit = new ThirdPartyTextUnit();
+      thirdPartyTextUnit.setTmTextUnit(tu);
+      thirdPartyTextUnit.setAsset(asset);
+      thirdPartyTextUnit.setUploadedFileUri("testPluralUploadedFileUri");
+      thirdPartyTextUnitRepository.save(thirdPartyTextUnit);
+    }
+
+    prepareAssetAndTextUnits(assetExtraction, asset, tm);
+
+    tmsSmartling.pushAITranslations(
+        repository,
+        "projectId",
+        pluralSep,
+        ImmutableMap.of(),
+        null,
+        null,
+        null,
+        ImmutableList.of());
+    result = resultProcessor.pushAITranslationFiles;
+
+    assertThat(result).hasSize(4);
+    Stream.of(jaJP, frCA)
+        .forEach(
+            locale -> {
+              SmartlingFile singularFile =
+                  result.stream()
+                      .filter(
+                          f ->
+                              f.getFileName()
+                                  .matches("testSingularUploadedFileUri_" + locale.getBcp47Tag()))
+                      .findFirst()
+                      .get();
+
+              SmartlingFile pluralFile =
+                  result.stream()
+                      .filter(
+                          f ->
+                              f.getFileName()
+                                  .matches("testPluralUploadedFileUri_" + locale.getBcp47Tag()))
+                      .findFirst()
+                      .get();
+
+              assertThat(singularFile.getFileName())
+                  .isEqualTo("testSingularUploadedFileUri_" + locale.getBcp47Tag());
+              List<TextUnitDTO> singulars = readTextUnits(singularFile, pluralSep);
+              assertThat(singulars)
+                  .allSatisfy(
+                      tu -> {
+                        assertThat(tu.getComment()).startsWith("Singular Comment");
+                        assertThat(tu.getName()).startsWith("singular_message");
+                        assertThat(tu.getAssetPath()).isEqualTo("fake_for_test");
+                        assertThat(tu.getTarget())
+                            .matches("Singular Message Test #\\d in " + locale.getBcp47Tag());
+                      });
+
+              assertThat(pluralFile.getFileName())
+                  .isEqualTo("testPluralUploadedFileUri_" + locale.getBcp47Tag());
```

Comment:
Ah - maybe related to my prior comment re filename construction - if we had that in a different name - we could reuse it here? (more of a nit than anything)